docker/dockerfiles/Dockerfile

```diff
@@ -132,22 +133,29 @@ WORKDIR /tmp
 # en_US.UTF-8 locale not available by default
 RUN locale-gen en_US.UTF-8
 
-# install mysql
+# Install MySQL.
+#
+# We explicitly target MySQL 5.7, and because of that we also explicitly target
+# Ubuntu Bionic (18.x).
+#
+# We also explicitly install both mysql-client and mysql-server, so that we can
+# also explicitly define the versions of both; normally only the latter is
+# required since it depends on the former.
+#
+# Once we update to MySQL 8+, this can be simplified.
 RUN curl -sSLOJ https://dev.mysql.com/get/mysql-apt-config_0.8.22-1_all.deb && \
     echo "mysql-apt-config mysql-apt-config/select-server select mysql-5.7" | /usr/bin/debconf-set-selections && \
+    echo "mysql-apt-config mysql-apt-config/repo-codename select bionic" | /usr/bin/debconf-set-selections && \
     dpkg -i mysql-apt-config_0.8.22-1_all.deb || apt-get -fy install && \
     rm -rf mysql-apt-config_0.8.22-1_all.deb && \
     apt-get update && \
-    # warning: mysql-server debconf hangs forever if wget is already installed (!)
-    apt-get install -y mysql-server libmysqlclient-dev
+    apt-get install --yes libmysqlclient-dev mysql-client=5.7.42-1ubuntu18.04 mysql-server=5.7.42-1ubuntu18.04
 
 # disable auth for local dashboard mysql db
-RUN chown -R mysql:mysql /var/lib/mysql && \
-    service mysql start && \
+RUN service mysql start && \
     echo "ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';" | mysql && \
     echo "CREATE USER 'readonly'@'localhost' IDENTIFIED WITH mysql_native_password BY '';" | mysql && \
     echo "GRANT SELECT ON dashboard_test.* TO 'readonly'@'localhost';" | mysql && \
-    chown -R mysql:mysql /var/lib/mysql /var/run/mysqld && \
```

Comment:
It's not clear to me why we were doing this before (nor, it seems, [was it clear to the last folks to find themselves here](https://github.com/code-dot-org/code-dot-org/pull/29503#discussion_r302259709)), but it no longer seems to be required and starting with Ubuntu 20 it causes some problems. So, I propose we remove it for the same reason as we added it in the first place: doing so make this work.